src/message_pool.zig

```diff
@@ -123,7 +123,7 @@ pub const MessagePool = struct {
         var free_count: usize = 0;
         while (pool.free_list) |message| {
             pool.free_list = message.next;
-            allocator.free(message.buffer);
+            allocator.free(@as([]const u8, message.buffer));
```

Comment:
Did this sneak in by mistake (rebase issue maybe)? https://github.com/tigerbeetle/tigerbeetle/pull/1140